aws_lambda_powertools/utilities/feature_toggles/configuration_store.py

```diff
@@ -0,0 +1,207 @@
+# pylint: disable=no-name-in-module,line-too-long
+from enum import Enum
+from typing import Any, Dict, List, Optional
+
+from botocore.config import Config
+
+from aws_lambda_powertools.utilities.parameters import AppConfigProvider, GetParameterError, TransformParameterError
+
+TRANSFORM_TYPE = "json"
+FEATURES_KEY = "features"
+RULES_KEY = "rules"
+DEFAULT_VAL_KEY = "default_value"
+RESTRICTIONS_KEY = "restrictions"
+RULE_NAME_KEY = "name"
+RULE_DEFAULT_VALUE = "default_value"
+RESTRICTION_KEY = "key"
+RESTRICTION_VALUE = "value"
+RESTRICTION_ACTION = "action"
+
+
+class ACTION(str, Enum):
+    EQUALS = "EQUALS"
+    STARTSWITH = "STARTSWITH"
+    ENDSWITH = "ENDSWITH"
+    CONTAINS = "CONTAINS"
+
+
+class ConfigurationException(Exception):
+    pass
+
+
+class ConfigurationStore:
+    def __init__(
+        self, environment: str, service: str, conf_name: str, cache_seconds: int, region_name: Optional[str] = None
+    ):
+        """constructor
+
+        Args:
+            environment (str): what appconfig environment to use 'dev/test' etc.
+            service (str): what service name to use from the supplied environment
+            conf_name (str): what configuration to take from the environment & service combination
+            cache_seconds (int): cache expiration time, how often to call AppConfig to fetch latest configuration
+            region_name (str): aws region where the configuration resides in
+        """
+        self._cache_seconds = cache_seconds
+        self._conf_name = conf_name
+        config = None
+        if region_name is not None:
+            config = Config(region_name=region_name)
+        self._conf_store = AppConfigProvider(environment=environment, application=service, config=config)
+
+    def _validate_json_schema(self, schema: str) -> bool:
+        ## todo
+        return True
+
+    def _match_by_action(self, action: str, restriction_value: Any, context_value: Any) -> bool:
+        mapping_by_action = {
+            ACTION.EQUALS.value: lambda a, b: a == b,
+            ACTION.STARTSWITH.value: lambda a, b: a.startswith(b),
+            ACTION.ENDSWITH.value: lambda a, b: a.endswith(b),
+            ACTION.CONTAINS.value: lambda a, b: a in b,
+        }
+
+        try:
+            func = mapping_by_action.get(action, lambda a, b: False)
+            return func(context_value, restriction_value)
+        except Exception:
+            return False
+
+    def _handle_rules(
+        self,
+        logger: object,
+        feature_name: str,
+        rules_context: Dict[str, Any],
+        default_value: bool,
+        rules: List[Dict[str, Any]],
+    ) -> bool:
+        for rule in rules:
+            rule_name = rule.get(RULE_NAME_KEY, "")
+            rule_default_value = rule.get(RULE_DEFAULT_VALUE)
```

Comment:
Shouldn't this and similar get operations all handle missing/default values?

Reply:
yeah, so basically i added a schema validation function so i wouldnt need to do all these checks here. In my original solution i used Pydantic to parse the input. But Heitor told me Pydantic is not an option here since it's a very large dependancy. How do you suggest I validate the schema? maybe similar to the validator utility?

Reply:
I've commented below but I like the idea of keeping a simple function at the moment with ```if name in object``` conditions which can be extended in the future with the validator if needed without breaking any external interfaces